### PR TITLE
Feat/ocp stream plugins

### PR DESCRIPTION
**Pull request by JarbasAl**
do not hardcode only a few stream extractors, move them to external plugins

NOTE: the objective of plugins is to move the stream extraction step to the playback step, instead of requiring it while searching for results, skills can request specific extractors via specific uris, additionally extractors can handle regular urls and determine if they can handle them

plugins:
- https://github.com/OpenVoiceOS/ovos-ocp-files-plugin
- https://github.com/OpenVoiceOS/ovos-ocp-youtube-plugin
- https://github.com/OpenVoiceOS/ovos-ocp-m3u-plugin
- https://github.com/OpenVoiceOS/ovos-ocp-rss-plugin
- https://github.com/OpenVoiceOS/ovos-ocp-deezer-plugin
- https://github.com/OpenVoiceOS/ovos-ocp-bandcamp-plugin
- https://github.com/OpenVoiceOS/ovos-ocp-news-plugin

### CODE REVIEW

File: ovos_plugin_common_play/ocp/utils.py [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+    """ Determine mime type. """
+    mime = mimetypes.guess_type(uri)
+    if mime:
+        return mime
+    else:
+        return None
+
+
+def available_extractors():
+    return ["/", "http"] + \
```
Comment (NeonDaniel) [inline ovos_plugin_common_play/ocp/utils.py:23]: I'd like to make this more explicit as `file://`, `http://`, `https://` if we're just looking at string prefixes; we can handle parsing `/` to `file://` pretty easily and this makes it easier to handle more paths like `smb://`, `nfs://`, etc.

File: ovos_plugin_common_play/ocp/utils.py [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+           [f"{sei}//" for sei in ocp_plugins.supported_seis]
```
Comment (NeonDaniel) [inline ovos_plugin_common_play/ocp/utils.py:24]: should this be `{sei}://`? Not sure how network or local file paths expand without the `:`

Comment (JarbasAl) [reply ovos_plugin_common_play/ocp/utils.py:24]: skill in the wild are already using the old scheme, so this is not so trivial to change. but this is all stripped by OCP itself before playback so not sure what you mean by expand?

Comment (JarbasAl) [reply ovos_plugin_common_play/ocp/utils.py:24]: i think there was an issue somewhere in the pipeline splitting at `:` and thats why i didnt add it originally, but i don't recall details....

**Pull request merged**